File: src/main/scala/com/teambr/modularsystems/storage/blocks/StorageState.scala
package com.teambr.modularsystems.storage.blocks

import java.util

import com.google.common.collect.ImmutableMap
import net.minecraft.block.Block
import net.minecraft.block.properties.IProperty
import net.minecraft.block.state.IBlockState
import net.minecraft.util.BlockPos
import net.minecraft.world.IBlockAccess

/**
 * This file was created for Modular-Systems
 *
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 08, 2015
 */
class StorageState(val connected : Boolean, val pos : BlockPos,  val world : IBlockAccess, val block : Block) extends IBlockState {
    override def getBlock : Block = block
    override def withProperty(property : IProperty, value : Comparable[_]) : IBlockState = null
    override def getValue(property : IProperty) : Comparable[_] = null
    override def cycleProperty(property : IProperty) : IBlockState = null
    override def getPropertyNames : util.Collection[_] = null
    override def getProperties : ImmutableMap[_, _] = null
}

File: src/main/scala/com/teambr/modularsystems/storage/blocks/StorageState.java
package com.teambr.modularsystems.storage.blocks;

import com.google.common.collect.ImmutableMap;
import net.minecraft.block.Block;
import net.minecraft.block.properties.IProperty;
import net.minecraft.block.state.IBlockState;
import net.minecraft.util.BlockPos;
import net.minecraft.world.IBlockAccess;

import java.util.Collection;

/**
 * This file was created for Modular-Systems
 * <p/>
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since 1/17/2016
 */
public class StorageState implements IBlockState {
    public boolean connected;
    public BlockPos pos;
    public IBlockAccess world;
    public Block block;

    public StorageState(boolean c, BlockPos p, IBlockAccess w, Block b) {
        connected = c;
        pos = p;
        world = w;
        block = b;
    }

    @Override
    public Collection<IProperty> getPropertyNames() {
        return null;
    }

    @Override
    public <T extends Comparable<T>> T getValue(IProperty<T> property) {
        return null;
    }

    @Override
    public <T extends Comparable<T>, V extends T> IBlockState withProperty(IProperty<T> property, V value) {
        return null;
    }

    @Override
    public <T extends Comparable<T>> IBlockState cycleProperty(IProperty<T> property) {
        return null;
    }

    @Override
    public ImmutableMap<IProperty, Comparable> getProperties() {
        return null;
    }

    @Override
    public Block getBlock() {
        return block;
    }
}
